### DISCUSSION TITLE
Is there a way to get server port via an extension?

### DISCUSSION THREAD

**Discussion by emortaldev** [Minestom/Minestom#789] [Help]
.

**Comment by emortaldev** [comment]
`MinecraftServer.getServer().getPort()`

**Reply by TheMode** [reply]
Be aware that this may not be usable depending on the socket type (e.g. unix domain)

**Reply by emortaldev** [reply]
Yup it's only returning 0, is there something else I can use?

**Reply by BomBardyGamer** [reply]
If you're using a UNIX domain socket, there is no port, as UNIX domain sockets use files to transfer data between processes. The only time you'll be using a UNIX domain socket though is when you are using a proxy, in which you'd have to get the port from the proxy, since Minestom doesn't know it.

**Comment by BomBardyGamer** [comment]
If you're using Minestom standalone, without a proxy, or you're using it with a proxy but using TCP/IP to communicate with the proxy, then you can get the port from the socket `Server`, which should be available through `MinecraftServer.getServer().getPort()`, as was pointed out. However, if you're using a proxy and communicating via UNIX domain socket, there is no port, as it's using a file to communicate, in which the port [will always be `0`](https://github.com/Minestom/Minestom/blob/3fbdf4f22a0a4c742256219b8375f27e52523a31/src/main/java/net/minestom/server/network/socket/Server.java#L54). If you want to get the port in that case, you probably want the port that the proxy is bound to that is exposed to the internet, which Minestom doesn't know, therefore you have two options:
1. Get it from the proxy using a proxy plugin
2. Find a different way to do what you're trying to do which doesn't need the port, and/or in some other way supports UNIX domain sockets
3. Throw in the towel and accept defeat

Just thought I'd put all the information in one place.
